### PR TITLE
Validate crs

### PR DESCRIPTION
**Pull request by mdsgabriel**
*Issue #, if available:*

*Description of changes:*
This will verify if the following objects exist before creating a workload cluster

1. Cluster
2. GitOpsConfig
3. VsphereMachineConfig
4. OIDCCOnfig
5. EKS Anywhere CRD

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/537#" title="Author self-approved">mdsgabriel</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [mdsgabriel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/providers/vsphere/vsphere.go
Comment (danbudris) [inline pkg/providers/vsphere/vsphere.go:871]: could we check all of the MachineConfigs for duplicate names and then return an error the indicates those MCs which are duplicates, rather than returning an error on the first duplicate we hit? This way multiple duplicate machine configs can be fixed by the user all at once.

File: pkg/executables/kubectl.go
Comment (danbudris) [inline pkg/executables/kubectl.go:413]: maybe these methods should be called something like `EksaClusterCrdExists` (or for the above `ValidateClustersCRD` could be `CapiClusterCrdExists`)? `Validate` seems a bit overloaded.

**Review by danbudris — APPROVED** [review]

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/537#pullrequestreview-794420438" title="Approved">danbudris</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/537#" title="Author self-approved">mdsgabriel</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [danbudris,mdsgabriel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/providers/vsphere/vsphere.go
Comment (jiayiwang7) [reply pkg/providers/vsphere/vsphere.go:871]: I'm thinking if we could reuse the `GetEksaVSphereMachineConfig` instead of `Searchxxx` and just add an error message type of `ResourceNotFound`. So that we don't replicate all the Get methods.

Comment (jiayiwang7) [reply pkg/providers/vsphere/vsphere.go:871]: Also, what do u think abt moving these checks to the preflight validation instead of here?

**Review by jiayiwang7 — APPROVED** [review]

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/537#pullrequestreview-794420438" title="Approved">danbudris</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/537#pullrequestreview-794435793" title="Approved">jiayiwang7</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/537#" title="Author self-approved">mdsgabriel</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [danbudris,jiayiwang7,mdsgabriel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/providers/vsphere/vsphere.go
Comment (jiayiwang7) [reply pkg/providers/vsphere/vsphere.go:871]: chatted with @mdsgabriel offline, gonna work on it in a separate PR. (https://github.com/aws/eks-anywhere/issues/539)

**Pull request merged**
